pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -404,6 +405,11 @@ func (ctx *context) getDIDDocAndConnection(pubDID string) (*did.Doc, *Connection
 			return nil, nil, fmt.Errorf("resolve public did[%s]: %w", pubDID, err)
 		}
 
+		err = ctx.connectionStore.didMap.SaveDIDFromDoc(didDoc, didCommServiceType, ed25519KeyType)
```

Comment:
3 references above the func call might call for a refactor.

Reply:
I will make an issue for this one - (permanent) connectionStore and the DIDConnection Store should be merged in another PR.

Reply:
#1004